sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -72,6 +76,23 @@ case class GpuUnaryMinus(child: Expression, failOnError: Boolean) extends GpuUna
       // Because of 2s compliment we need to only worry about the min value for integer types.
       GpuAnsi.assertMinValueOverflow(input, "minus")
     }
+
+    if (GpuTypeShims.isDayTimeInterval(dataType)) {
+      // For day time interval, Spark throws an exception when overflow,
+      // regardless of whether `SQLConf.get.ansiEnabled` is true or false
+      withResource(Scalar.fromLong(Long.MinValue)) { minVal =>
+        GpuAnsi.assertMinValueOverflowImp(minVal, input, "minus")
+      }
+    }
+
+    if (GpuTypeShims.isYearMonthInterval(dataType)) {
+      // For day time interval, Spark throws an exception when overflow,
+      // regardless of whether `SQLConf.get.ansiEnabled` is true or false
+      withResource(Scalar.fromLong(Int.MinValue)) { minVal =>
```

Comment:
Is it expected that creating a long scalar from an int value ?

Reply:
done